File: Technology Transfer _ Sulambi VMS/Source Code/sulambi-backend-main/sulambi-backend-main/app/routes/feedback.py
from flask import Blueprint, request
from ..controllers import feedback

FeedbackBlueprint = Blueprint('feedback', __name__, url_prefix="/feedback")

@FeedbackBlueprint.get("/<eventType>/<eventId>")
def getFeedback(eventType: str, eventId: int):
    return feedback.getEventFeedback(eventType, eventId)

@FeedbackBlueprint.post("/<eventType>/<eventId>")
def createFeedback(eventType: str, eventId: int):
    if (eventType not in ["external", "internal"]):
        return ({ "message": "Invalid event type" }, 400)
    return feedback.createFeedback(eventType, eventId, request.json.get("feedback"))

@FeedbackBlueprint.put("/<feedbackId>")
def updateFeedback(feedbackId: int):
    userFeedback = request.json.get("feedback")
    return feedback.updateFeedback(feedbackId, userFeedback)

File: SulambiVosaMain/Technology Transfer _ Sulambi VMS/Source Code/sulambi-frontend-main/sulambi-frontend-main/src/api/feedback.ts
import axios from "./init";

const basePath = "/feedback";

export const getEventFeedback = async (
  eventType: "external" | "internal",
  eventId: number
) => {
  return axios.get(`${basePath}/${eventType}/${eventId}`);
};

export const submitFeedback = async (
  eventType: "external" | "internal",
  eventId: number,
  feedback: string
) => {
  return axios.post(`${basePath}/${eventType}/${eventId}`, { feedback });
};

export const updateFeedback = async (feedbackId: number, feedback: string) => {
  return axios.put(`${basePath}/${feedbackId}`, { feedback });
};
